source/modules/sensors/SensorsGetters.mc

```diff
@@ -268,16 +268,24 @@ module SensorsGetters {
             return System.getDeviceSettings().phoneConnected;
         }
 
-        function getHR() as Number? {
+        function getHR() as Array<Number?>? {
             var activityInfo = Activity.getActivityInfo();
             var value = activityInfo != null ? activityInfo.currentHeartRate : null;
-
+            var isHeartWorking = true;
             if (value == null && Checkers.checkHeartRateHistory()) {
                 var sensorInfo = SensorHistory.getHeartRateHistory({}).next();
                 value = sensorInfo != null ? sensorInfo.data : value;
+                if (value != null) {
+                    isHeartWorking = false;
```

Comment:
Here and below too ;)

Reply:
Have a good rest, and I will do the rest. Also, use 1 if isHeartWorking is true, otherwise use 0.